code/modules/clothing/clothing.dm

```diff
@@ -120,10 +142,15 @@
 			right_overlay.alpha = dye_alpha
 			dyn_overlay_right.overlays += right_overlay
 
+	if ((luminosity > 0) || (dyed_parts.len > 0))
 		dynamic_overlay["[cloth_layer]"] = dyn_overlay_worn
 		dynamic_overlay["[HAND_LAYER]-[GRASP_LEFT_HAND]"] = dyn_overlay_left
 		dynamic_overlay["[HAND_LAYER]-[GRASP_RIGHT_HAND]"] = dyn_overlay_right
 
+	set_blood_overlay()//re-applying blood stains
+	if (on_fire && fire_overlay)
+		overlays += fire_overlay
```

Comment:
Mentioned this on your other PR, but I don't think this is necessary.

Reply:
same thing here. it is necessary.